src/engine/enginemaster.cpp

```diff
@@ -366,16 +406,80 @@ void EngineMaster::process(const int iBufferSize) {
     // Mix all the PFL enabled channels together.
     m_headphoneGain.setGain(chead_gain);
 
-    if (m_bRampingGain) {
-        ChannelMixer::mixChannelsRamping(
-                m_headphoneGain, &m_activeHeadphoneChannels,
-                &m_channelHeadphoneGainCache,
-                m_pHead, iBufferSize);
+    // Mixxx gets input signals delayed by the input latency,
+    // so to preserve the relative timing of the input with the master and headphones
+    // outputs, delay the master, booth, and headphones outputs by the input latency.
+    // This way, if the microphone user is on beat with the output they hear coming from
+    // the speakers/headphones, the input will remain on beat at the expense of
+    // adding some latency.
```

Comment:
> The input latency compensation delays only apply to the master mix routed to each output.

This will not work, because the user must not hear the delay compensated signal. He has to do all actions, and his voice along with not delayed signals. If he hears the delayed signal on his headphone, he will speak too late and the compensation has no effect.